### ISSUE TITLE
Unit test with Table-driven style

### ISSUE THREAD

**Issue by Yisaer** [pingcap/pd#2590]
## Feature Request

In current unit test coverage situation, I found that, in most cases, the error handling hasn't been covered due to we only prepared the success cases in current unit test.

Here is one example:

![image](https://user-images.githubusercontent.com/13427348/86008744-c8a82e00-ba4b-11ea-92d3-d30fb06b8afa.png)

To cover the error handling in the function, I'd like to introduce the [`Table-driven`](https://github.com/golang/go/wiki/TableDrivenTests#parallel-testing)  ut style. 

Here is one example to show how we use table-driver style to cover the error handling cases. 

https://github.com/pingcap/pd/pull/2589

**Comment by lhy1024** [comment]
Good idea. There are some Table-driven cases in test cases, but not all.
